pep-0708.rst

```diff
@@ -0,0 +1,835 @@
+        },
+        {
+          "paths": ["*"],
+          "repositories": ["PyPI"]
+        }
+      ]
+    }
+
+The recommendation to have explicit configuration pushes the decision on how to
+implement that onto each installer, allowing them to choose what works best for
+their users.
+
+Ultimately only implementing some kind of explicit configuration was rejected
+because by its nature it's opt in, so it doesn't protect average users who are
+least capable to solve the problem with the existing tools, by adding additional
+protections alongside the explicit configuration, we are able to protect all
+users by default.
+
+Additionally, relying on only explicit configuration also means that every end
+user has to resolve the same problem over and over again, even in cases like
+mirrors of PyPI, Piwheels, PyTorch, etc. In each and every case they have to sit
+there and make decisions (or find some example to cargo cult) in order to be
+secure. Adding extra features into the mix allows us to centralize those
+protections where we can, while still giving advanced end users the ability to
+completely control their own destiny.
+
+
+Scopes à la npm
+---------------
+
+There's been some suggestion that scopes similar to how npm has implemented them
+may ultimately solve this. Ultimately scopes do not change anything about this
+problem. As far as I know scopes in npm are not globally unique, they're tied to
+a specific registry just like unscoped names are. However what scopes do enable
+is an obvious mechanism for grouping related projects and the ability for a user
+or organization on npm.org to claim an entire scope, which makes explicit
+configuration significantly easier to handle because you can be assured that
+there's a whole little slice of the namespace that wholly belongs to you, and
+you can easily write a rule that assigns an entire scope to a specific non
+public registry.
+
+Unfortunately, it basically ends up being an easier version of the idea to only
+use explicit configuration, which works ok in npm because its not particularly
+common for people to use their own registries, but in Python we encourage you to
+do just that.
+
+
+Open questions
+==============
+
+1. The `original proposal document <https://docs.google.com/document/d/184fQkb6NggVQfYmjTDA7p_U3iWDKk6grc2DigT1X3Es/>`__
+   was targeted more specifically to a change to pip, and went into more
+   specific details as to what we expected from pip. Since dictating UX to
+   installers isn't something that we do in PEPs, I've rewritten those parts to
+   be more generic, however that means that we lose the information on
+   repository files. Is that fine? Or should we standardize what a repository
+   file looks like so the same file can be given to multiple installers instead
+   of hand waving around the specific mechanism installers would use for
+   explicit configuration?
+2. Is the Appendix section on communicating the change useful or confusing?
```

Comment:
For what it's worth, I think it's useful. It actually fits well with the "How to Teach This" section in the standard PEP template.

Reply:
Yea, I tend to lean towards it being helpful, my main concern was it side tracking discussion or random people finding it and thinking it's an actual announcement of some kind, but I left the question open in case someone felt strongly about it being too specific to pip.

Reply:
I find it helpful, and you could include it as part of the "How to Teach This" section in this PEP. Alternatively, if you want to separate it from the main PEP content, you could put it into a separate `rst` document in the `pep-0708/` directory and use a ref target label to link it here instead.

Reply:
I've moved the Appendix to a "How to Communicate This" section and removed the open question. Thanks!